### PR TITLE
Make futures-extra compatible with Guava 20.

### PR DESCRIPTION
**Pull request by rydenius**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.spotify</groupId>
-  <version>2.6.3-SNAPSHOT</version>
+  <version>3.0.0-SNAPSHOT</version>
```
Comment (dflemstr) [inline pom.xml]: Why the version bump?  There are no breaking changes as far as I can tell.

Comment (TheIndifferent) [inline pom.xml]: `Futures.transformAsync` appeared only in Guava 19, stated compatibility of current version is Guava 18.

Comment (rydenius) [inline pom.xml]: Yes, I thought I had to do some kind of version bump to signal that Guava requirements had changed. But maybe 3.0 was over the top. Bumped to 2.7 now instead. Ok?

File: pom.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.spotify</groupId>
-  <version>2.6.3-SNAPSHOT</version>
+  <version>2.7.0-SNAPSHOT</version>
```
Comment (dflemstr) [inline pom.xml]: No signatures changed, so this should still be `2.6.3`... However I'll still merge this if you don't want to fix this :)

Comment (rydenius) [inline pom.xml]: I still think the need for Guava major bump should be reflected somehow in our version numbering. In addition I deprecated a public method.

File: README.md [hunk 3]
```diff
@@ -71,6 +71,9 @@ This is just a simple delegating method that explicitly calls
 Futures.transform(future, Function). There is also a corresponding
```
Comment (pettermahlen) [inline README.md]: This line doesn't seem to be correct any longer, IIUC? The call is instead to `Futures.transformAsync`?

Comment (rydenius) [inline README.md]: No that is correct. It is `Futures.transform(future, Function)` but `Futures.transformAsync(future, AsyncFunction)`. They used to have `Futures.transform` overloaded with both Function and AsyncFunction arguments. In Guava 19 they added transformAsync and deprecated the transform variant that took AsyncFunction in preparation for removing it in Guava 20.

Comment (pettermahlen) [inline README.md]: Got it.

**Pull request merged**